### DISCUSSION TITLE
plugin development for sumatrapdf

### DISCUSSION THREAD

**Discussion by thegoder66** [sumatrapdfreader/sumatrapdf#3068] [Ideas]
Hi,
many software support a plugin SDK to extend its functionality. can it be done in sumatrapdf?

**Comment by GitHubRulesOK** [comment]
possibly, since in addition to others, the Chinese sell a SumatraPDF clone that is drastically enhanced, but they employ dozens of programmers on their pirate copy. it has cropping, page arranger, signing, merging, short cut editor, file conversion, and all-round is a good Acrobat alternative editing tool. However I did not see any plugin system.

Closest I can do is use other open source external tools to convert/rotate pages etc.

**Reply by piyush1104** [reply]
Do you have somehow the link to the chinese clone?

**Reply by linkouts** [reply]
It's probably added to the software ,like Foxit PDF Editor, PDF-XChange Editor.
Adding these functions to sumatrapdf is not practical

**Reply by GitHubRulesOK** [reply]
@linkouts 
its a standard copy of SumatraPDF inside their Interface (not for the first time have seen similar), however unlike Russian HamsterPDF or VeryPDF DRM Reader PDFlite or many other with lightweight changes this one takes bolt-ons to a highly sophisticated level

**Comment by kjk** [comment]
No plans for plugins. They're way too complicated and they are not magic. First step is to list the kinds of things you would like to see developed. Then we can talk how this could be achieved and if plugins would even work.

**Reply by Foadsf** [reply]
For example I would like to have a "AI chat with PDF" plugin. There many possibilities, either using APIs, or Ollama... implementing LLM RAG would be great.

**P.S.** Also text to speech (TTS) could be a nice to have feature.

**Reply by GitHubRulesOK** [reply]
@Foadsf 
SumatraPDF can send current focus to any web or internal application at the press of a single key, but you need to know what that external app requires so send a selected sentence to a web AI or file page number to TTS like [Balcon /Balabolka](https://www.cross-plus-a.com/balabolka.htm) is not a problem.

For example just answered how to send Pdf to MS edge for print but could include go to page number `--app="%1#page=%p"`

https://github.com/sumatrapdfreader/sumatrapdf/issues/4611#issuecomment-2451685869

**Comment by GitHubRulesOK** [comment]
SumatraPDF 3.7 has inbuilt tools and thus no need for online tools all functions should be potentially possible OFFline

However AI has no real ability to edit or read PDF encodings as it is a printed Viewing Format unsuited to input task such as TTS which is TEXT and thus export as test is just as usable to AI and TTS
